senders/senders.py

```diff
@@ -9,8 +9,16 @@ class Sender:
     def send(self, providers):
         raise Exception('Please implement this method.')
 
+    def scheduleCheck(self, ev, unit):
+        if unit not in ['seconds', 'minutes', 'hours', 'days']:
+            raise Exception('Bad scheduling information provided')
+        else:
+            self.schedule_ev = ev
+            self.schedule_unit = unit
+
 class MailSender(Sender):
-    def __init__ (self, smtp_server, sender_address, recipient_address):
+    def __init__ (self, ev, unit, smtp_server, sender_address, recipient_address):
+        self.scheduleCheck(ev,unit)
```

Comment:
Add the correct `__init__` function in the `Sender` class, and use the [`super()`](https://docs.python.org/3/library/functions.html#super) builtin instead of your `scheduleCheck` function :) (the exact syntax will be something like `super().__init__(args)`).